Sources/Sentry/SentryProfiler.mm

```diff
@@ -721,7 +738,6 @@ - (void)stop
     }
 
     _profiler->stopSampling();
-    _endDate = [SentryCurrentDate date];
```

Comment:
Can't comment on the line directly but there's a line below that says
```objc
profile[@"timestamp"] = [[SentryCurrentDate date] sentry_toIso8601String];
```
where you set the timestamp on the profile. If I'm reading it right, this is setting the timestamp on the profile to be the current time when the profile is being serialized right? So the adjustments we'll be doing on the UI will be based on this timestamp. So each sample is assumed to have occurred at the profile's timestamp + elapsed since start ns. And ideally, the profile's timestamp == transaction's start timestamps which is what I assume you're doing in this PR (I didn't read the rest)

Reply:
I'm pretty sure I remember this was supposed to be the time that we were sending the profile, and the timestamp for the first sample would be used for the start, but sadly that convo is long gone in slack, and isn't on GitHub. We can easily change this is that is not the case.

Reply:
@phacops can you comment on whether it would break anything if we changed this assumption, or is the assumption incorrect right now?

Reply:
in any case we can handle this in a separate PR

Reply:
You can change the assumption now, we never really used that field. It was meant to be something different indeed when we had support for multiple transactions, now it's best to be the reference time from which `elapsed_since_start_ns` are relative to.

Also, a maximum amount of precision is best, we support `RFC3339Nano` format, like `2023-03-01T10:00:00.123456789-07:00`.

Reply:
In that case can we remove it altogether since all times will be relative to the transaction start which is reported in the main envelope payload, or is it not possible to grab that?

I’ll have to check on RFC3339 support in iOS, I know there’s a builtin way to get three sigfigs on the seconds fraction for milliseconds, need to check about 9 for nano.

Reply:
Actually, the best value for you would be exactly what you used for the start of the transaction at the same precision.

In the backend, we'll still use that field to do `timestamp - start of the transaction` and adjust relative timestamps, but in this case, it should be equal to 0 and no adjustment will take place.

Reply:
Please double check that I've captured the requirement correctly in https://github.com/getsentry/team-profiling/issues/191 @phacops @Zylphrex